ape_hardhat/provider.py

```diff
@@ -377,10 +432,13 @@ def _set_web3(self):
         client_version = self._web3.client_version.lower()
         if "hardhat" in client_version:
             self._web3.eth.set_gas_price_strategy(rpc_gas_price_strategy)
-        else:
-            raise ProviderError(
-                f"Port '{self.port}' already in use by another process that isn't a Hardhat node."
+        elif self._port is not None:
+            raise HardhatProviderError(
+                f"Port '{self._port}' already in use by another process that isn't a Hardhat node."
```

Comment:
This log line is not super helpful when connecting to remote hosts.
Let's change it to `A process that is not a Hardhat node is running at host {self._clean_uri}`.